CMakeLists.txt

```diff
@@ -32,11 +32,18 @@ IF(WIN32)
     add_definitions(-D_CRT_SECURE_NO_WARNINGS)
     # Make warning as error
     add_definitions(/WX)
-ELSE()
+ELSEIF(APPLE)
     # Make warning as error
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+ELSE()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
-ENDIF(WIN32)
+ENDIF()
+
+IF(APPLE)
+    SET(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
```

Comment:
OK, so the Python tools have a requirement that the thing they load must have an .SO suffix, whether it's a Mach-O shared library or a Mach-O module. I assume Python will dynamically load it with dlopen? In that case it really could be a shared library or a module. But other projects (e.g. the [gateway SDK](https://github.com/Azure/azure-iot-gateway-sdk)) need to be able to **link against** the IoT Hub device client library, so it can't be a module; it _has_ to be a shared library.

By convention, shared libraries have a DYLIB extension, and I'd rather not rename all the libraries in this repo to .SO just to make the Python tools happy.

Over in the Python repo (I see that you have an open pull request over there), why not set `CMAKE_SHARED_LIBRARY_SUFFIX` in the root CMakeLists.txt file, right before the call to  `add_subdirectory(c)`? Then all shared libraries in this repo will be renamed to .SO, but _only_ in the context of the Python SDK.